includes/components/class-attribute.php

```diff
@@ -307,6 +310,72 @@ public function register_models() {
 		}
 	}
 
+	/**
+	 * Validates model attributes.
+	 *
+	 * @param array  $errors Form errors.
+	 * @param object $form Form object.
+	 * @return array
+	 */
+	public function validate_model_attributes( $errors, $form ) {
```

Comment:
This callback is ok but a bit overkill, if you pass the "readonly" parameter to fields then they should take care of the rest of "readonly" logic by simply ignoring any values passed on form submission, if I remember correctly "readonly" fields are just ignored and unset so populating a custom array and comparing it is not required (please test this without a callback).

The field will be greyed out and it will not be possible to input anything anyway (with the "readonly" HTML attribute), and even if someone edits the source HTML code via browser dev tools the field value will be ignored on the back-end (this logic is already implemented), and showing errors to such users is not critical.

The task also doesn't mention custom errors, please always choose the minimum possible solution that matches the task requirements, I don't know if you discussed this one with Yevhen though

Reply:
this solution is due to the fact that after changing the value on the front - the form page is not reloaded and the user can endlessly change the value of the fields